ios/RCTWKWebView/RCTWKWebView.m

```diff
@@ -286,7 +286,14 @@ - (BOOL)canGoForward
 
 - (void)reload
 {
-  [_webView reload];
+  // fix reload problem after network connection failed
+  NSURLRequest *request = [RCTConvert NSURLRequest:self.source];
+
+  if (request.URL && !_webView.URL.absoluteString.length) {
+      [_webView loadRequest:request];
+  } else {
+      [_webView reload];
+  }
 }
```

Comment:
I'm concerned this violates the semantics of `reload`, which is documented both in React Native's `WebView` and in `WKWebView` as "reloads the current page".

This code might reload either the current page, or the page that the webview last had its `source` set to.  I may be misreading, but I think these could be different backs, if the user interacted with the page and navigated elsewhere since `source` was set?

I think maybe a better way to deal with a failed page load is to hook into `onError` in JavaScript and reload the page there?

Reply:
I agree. The `reload` method should be a simple bridged call to the native reload. We should avoid adding more logic here.